File: BOJ/11723_집합.py
import sys

bits = 0    # set S

M = int(sys.stdin.readline())
for _ in range(M):
  cmd = sys.stdin.readline().rstrip()

  if ' ' in cmd:                 # check command
    cmd1, x = cmd.split()        #     ..
  else:                          #     ..
    cmd1 = cmd                   #     ..

  if(cmd1 == "add"):
    bits |= (1<<int(x))

  elif(cmd1 == "remove"):
    bits &= ~(1<<int(x))

  elif(cmd1 == "check"):
    if(bits & (1<<int(x))):
      sys.stdout.write("1\n")
    else:
      sys.stdout.write("0\n")

  elif(cmd1 == "toggle"):
    bits ^= (1<<int(x))

  elif(cmd1 == "all"):      #    (1<<n) : n bit 가 1
    bits = ((1<<21) -1)     # (1<<n) -1 : n bit 아래 모든 bit 1

  else:
    bits = 0

File: BOJ/11723_집합.cpp
#include <iostream>
#include <string>
using namespace std;

int main(){
  ios_base::sync_with_stdio(false);     // cin cout 속도 향상
  cin.tie(NULL);                        //       ..
  cout.tie(NULL);                       //       ..

  int bits = 0;

  int M;
  cin >> M;
  while(M--){
    string cmd;
    int x;
    cin >> cmd;

    if(cmd == "add"){
      cin >> x;
      bits |= (1<<x);
    }
    else if(cmd == "remove"){
      cin >> x;
      bits &= ~(1<<x);
    }
    else if(cmd == "check"){
      cin >> x;
      if(bits & (1<<x))
        cout << "1\n";
      else
        cout << "0\n";
    }
    else if(cmd == "toggle"){
      cin >> x;
      bits ^= (1<<x);
    }
    else if(cmd == "all"){
      bits = ((1<<21) -1);    // (1<<n) -1 : n bit 아래 모두 1
    }
    else
      bits = 0;
  }
  return 0;
}
